src/types/routeProps.ts

```diff
@@ -1,15 +1,10 @@
-import { AsyncComponentLoader, Component, DefineComponent } from 'vue'
+import { Component } from 'vue'
 import { AfterRouteHook, BeforeRouteHook } from '@/types/hooks'
 import { Path } from '@/types/path'
 import { Query } from '@/types/query'
 import { Routes } from '@/types/route'
 import { MaybeArray } from '@/types/utilities'
 
-/**
- * Type alias for Vue components, which can be either synchronous or asynchronous components.
- */
-export type RouteComponent = Component | DefineComponent | AsyncComponentLoader
```

Comment:
I removed this type because I don't think we need it. I believe the built in `Component` type encompasses all of these types. All the tests and types passed without the extras.

Reply:
I'm all for simplicity if we can make things simpler. There will be some doc changes required if both our PRs get merged (note for future us)